gio/src/settings.rs

```diff
@@ -1,11 +1,115 @@
+        type Mappings = (
+            Option<Box<dyn Fn(&glib::Variant, glib::Type) -> Option<glib::Value>>>,
+            Option<Box<dyn Fn(&glib::Value, glib::VariantType) -> Option<glib::Variant>>>,
+        );
+        unsafe extern "C" fn bind_with_mapping_get_trampoline(
+            value: *mut glib::gobject_ffi::GValue,
+            variant: *mut glib::ffi::GVariant,
+            user_data: glib::ffi::gpointer,
+        ) -> glib::ffi::gboolean {
+            let user_data = &*(user_data as *const Mappings);
+            let f = user_data.0.as_ref().unwrap();
+            let value = &mut *(value as *mut glib::Value);
+            if let Some(v) = f(&*from_glib_borrow(variant), value.type_()) {
+                *value = v;
+                true
+            } else {
+                false
+            }
+            .to_glib()
+        }
+        unsafe extern "C" fn bind_with_mapping_set_trampoline(
+            value: *const glib::gobject_ffi::GValue,
+            variant_type: *const glib::ffi::GVariantType,
+            user_data: glib::ffi::gpointer,
+        ) -> *mut glib::ffi::GVariant {
+            let user_data = &*(user_data as *const Mappings);
+            let f = user_data.1.as_ref().unwrap();
+            let value = &*(value as *const glib::Value);
+            f(value, from_glib_none(variant_type)).to_glib_full()
+        }
+        unsafe extern "C" fn destroy_closure(ptr: *mut libc::c_void) {
+            Box::<Mappings>::from_raw(ptr as *mut _);
+        }
+        let get_trampoline: Option<unsafe extern "C" fn(_, _, _) -> _> =
+            if self.get_mapping.is_none() {
+                None
+            } else {
+                Some(bind_with_mapping_get_trampoline)
+            };
+        let set_trampoline: Option<unsafe extern "C" fn(_, _, _) -> _> =
+            if self.set_mapping.is_none() {
+                None
+            } else {
+                Some(bind_with_mapping_set_trampoline)
+            };
+        let mappings: Mappings = (self.get_mapping, self.set_mapping);
+        unsafe {
+            ffi::g_settings_bind_with_mapping(
+                self.settings.to_glib_none().0,
+                self.key.to_glib_none().0,
+                self.object.to_glib_none().0,
+                self.property.to_glib_none().0,
+                self.flags.to_glib(),
+                get_trampoline,
+                set_trampoline,
+                Box::into_raw(Box::new(mappings)) as *mut libc::c_void,
+                Some(destroy_closure),
+            )
+        }
+    }
+}
+
 pub trait SettingsExtManual {
     fn get<U: FromVariant>(&self, key: &str) -> U;
 
     fn set<U: ToVariant>(&self, key: &str, value: &U) -> Result<(), BoolError>;
+
+    #[doc(alias = "g_settings_bind")]
+    #[doc(alias = "g_settings_bind_with_mapping")]
+    fn bind<'a, P: IsA<glib::Object>>(
+        &'a self,
+        key: &'a str,
+        object: &'a P,
+        property: &'a str,
+        flags: SettingsBindFlags,
```

Comment:
I would move the flags into the builder and initialize with the default flags instead of having the user always provide flags

Reply:
Good point. Done.